engine/lib/phx/src/math/intersect.rs

```diff
@@ -83,14 +83,14 @@ pub unsafe extern "C" fn Intersect_PointTriangle_Barycentric(p: &Vec3, tri: &Tri
 }
 
 #[no_mangle]
-pub extern "C" fn Intersect_RayPlane(ray: &Ray, plane: &Plane, pHit: &mut Vec3) -> bool {
+pub extern "C" fn Intersect_RayPlane(ray: &Ray, plane: &Plane, pHit: &mut Position) -> bool {
     /* TODO : Shouldn't we handle denom == 0? */
-    let dist: f32 = (*plane).d - Vec3::dot((*plane).n, ray.p);
+    let dist: f32 = (*plane).d - DVec3::dot((*plane).n.as_dvec3(), ray.p.v) as f32;
     let denom: f32 = Vec3::dot((*plane).n, ray.dir);
     let t: f32 = dist / denom;
 
     if t >= ray.tMin && t <= ray.tMax {
-        *pHit = ray.p + ray.dir * t;
+        *pHit = Position::from_dvec(ray.p.v + (ray.dir * t).as_dvec3());
```

Comment:
DVec3 is also uses double. What is the difference with Position?

Reply:
Position is a first class type that communicates clearly its purpose, whilst DVec3 is the implementation detail of Position.